src/client/utils/getRouteData.ts

```diff
@@ -110,10 +110,9 @@ export default function getRouteData(
 				sections = [
 					{
 						...filteredSections[0],
-						components:
-							filteredComponents && targetIndex
-								? [filterComponentExamples(filteredComponents[0], targetIndex)]
-								: [],
+						components: filteredComponents
+							? [filterComponentExamples(filteredComponents[0], targetIndex as number)]
+							: [],
```

Comment:
I think something like this would be better:

```suggestion
						components: filteredComponents && typeof targetIndex === number
							? [filterComponentExamples(filteredComponents[0], targetIndex)]
							: [],
```

Avoid `as` because it basically disables type check.

Reply:
Done